types/topic_map.go

```diff
@@ -40,3 +40,15 @@ func (t *TopicMap) Sync(updated *map[string][]string) {
 
 	t.lookup = updated
 }
+
+func (t *TopicMap) Topics() []string {
+	t.lock.Lock()
```

Comment:
Can we use a RWMutex and RLock?

Reply:
Seems reasonable. Let me look into it and get back to you. Thanks!